### PR TITLE
chore(jellyfish-api-core): add more tests for poolSwap maxPrice and dex fees

### PR DESCRIPTION
**Pull request by fuxingloh**
#### What this PR does / why we need it:

Add more tests for poolSwap "maxPrice" when commissions and dex fees and added.

#### Which issue(s) does this PR fixes?:

Fixes #

### CODE REVIEW

**Review by ivan-zynesis — DISMISSED** [review]

File: packages/jellyfish-api-core/__tests__/category/poolpair/poolswap.test.ts [hunk 1]
```diff
@@ -136,6 +136,54 @@ describe('poolSwap', () => {
...
+    await container.generate(1)
+
+    const metadata = {
+      from: address,
+      tokenFrom: 'DFI',
+      amountFrom: 1,
+      to: address,
+      tokenTo: 'SLIP',
+      maxPrice: 1.5
+    }
```
Comment (surangap) [inline packages/jellyfish-api-core/__tests__/category/poolpair/poolswap.test.ts]:
```suggestion
    const inputDFIAmount = new BigNumber(1);
    const metadata = {
      from: address,
      tokenFrom: 'DFI',
      amountFrom: inputDFIAmount.toNumber(),
      to: address,
      tokenTo: 'SLIP',
      maxPrice: 1.5
    }

    // manual calculations and verify
    const inputDFIAfterFeeIn = inputDFIAmount.minus(inputDFIAmount.multipliedBy(0.05))
    // (SLIP 1000: DFI 1000)
    const ammSwappedAmountInSLIP = new BigNumber(1000).minus(new BigNumber(1000 * 1000).dividedBy(new BigNumber(inputDFIAfterFeeIn).plus(1000))).multipliedBy(100000000).minus(1).dividedBy(100000000).decimalPlaces(8, BigNumber.ROUND_CEIL)
    const finalAmountAfterFeeOut = ammSwappedAmountInSLIP.minus(ammSwappedAmountInSLIP.multipliedBy(0.3))
    expect(inputDFIAmount.dividedBy(finalAmountAfterFeeOut).gt(1.5)).toBeTruthy()
```

File: packages/jellyfish-transaction-builder/__tests__/txn/txn_builder_dex_compositeswap_maxprice.test.ts [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+      fromAmount: new BigNumber('1'),
+      toScript: script,
+      toTokenId: pairs.CAT.tokenId,
+      maxPrice: new BigNumber('1.2')
+    },
+    pools: [
+      { id: pairs.PIG.poolId },
+      { id: pairs.CAT.poolId }
+    ]
+  }, script)
```
Comment (surangap) [inline packages/jellyfish-transaction-builder/__tests__/txn/txn_builder_dex_compositeswap_maxprice.test.ts]:
```suggestion
   const inputPIGAmount = new BigNumber(1);
  const txn = await builder.dex.compositeSwap({
    poolSwap: {
      fromScript: script,
      fromTokenId: pairs.PIG.tokenId,
      fromAmount: inputPIGAmount,
      toScript: script,
      toTokenId: pairs.CAT.tokenId,
      maxPrice: new BigNumber('1.2')
    },
    pools: [
      { id: pairs.PIG.poolId },
      { id: pairs.CAT.poolId }
    ]
  }, script)

  // manual calculations and verify
  // PIG -> DFI
  const inputPIGAfterFeeIn = inputPIGAmount
  // (PIG 1000: DFI 1000)
  const ammSwappedAmountInDFI = new BigNumber(1000).minus(new BigNumber(1000 * 1000).dividedBy(new BigNumber(inputPIGAfterFeeIn).plus(1000))).multipliedBy(100000000).minus(1).dividedBy(100000000).decimalPlaces(8, BigNumber.ROUND_CEIL)
  const intermediateSwappedAmountAfterFeeOut = ammSwappedAmountInDFI.minus(ammSwappedAmountInDFI.multipliedBy(0.15))

  // DFI -> CAT
  const inputDFIAfterFeeIn = intermediateSwappedAmountAfterFeeOut.minus(intermediateSwappedAmountAfterFeeOut.multipliedBy(0.15))
  const ammSwappedAmountInCAT = new BigNumber(1000).minus(new BigNumber(1000 * 1000).dividedBy(new BigNumber(inputDFIAfterFeeIn).plus(1000))).multipliedBy(100000000).minus(1).dividedBy(100000000).decimalPlaces(8, BigNumber.ROUND_CEIL)
  const finalSwappedAmountAfterFeeOut = ammSwappedAmountInCAT;
  expect(inputPIGAmount.dividedBy(finalSwappedAmountAfterFeeOut).gt(1.2)).toBeTruthy()
```

**Review by surangap — COMMENTED** [review]
added manual calculation and verification.

**Comment by fuxingloh** [comment]
Thanks @canonbrother ❤️

**Comment by fuxingloh** [comment]
Merging this in now due to `stats-api` test fix.

**Pull request merged**